From 49236c9064e0a9a4b9d17d4f0b2acfe12b3f8657 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Fri, 12 Sep 2014 01:56:46 -0400
Subject: [drm] i915: split some DP modesetting code into a separate function

Message-id: <1410487019-24661-9-git-send-email-airlied@redhat.com>
Patchwork-id: 93921
O-Subject: [RHEL7.1 BZ1140440 PATCH 08/21] i915: split some DP modesetting code into a separate function
Bugzilla: 1140440
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Rob Clark <rclark@redhat.com>

bugzilla: 1140440
upstream: 44905a27dd01f2bddd96664d0ab7da43a07f4d5a

this is just prep work for mst support.

Signed-off-by: Dave Airlie <airlied@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_ddi.c b/drivers/gpu/drm/i915/intel_ddi.c
index a80cb3e..7a9186f 100644
--- a/drivers/gpu/drm/i915/intel_ddi.c
+++ b/drivers/gpu/drm/i915/intel_ddi.c
@@ -364,6 +364,17 @@ void hsw_fdi_link_train(struct drm_crtc *crtc)
  DRM_ERROR("FDI link training failed!\n");
 }
 
+void intel_ddi_init_dp_buf_reg(struct intel_encoder *encoder)
+{
+ struct intel_dp *intel_dp = enc_to_intel_dp(&encoder->base);
+ struct intel_digital_port *intel_dig_port =
+  enc_to_dig_port(&encoder->base);
+
+ intel_dp->DP = intel_dig_port->saved_port_bits |
+  DDI_BUF_CTL_ENABLE | DDI_BUF_EMP_400MV_0DB_HSW;
+ intel_dp->DP |= DDI_PORT_WIDTH(intel_dp->lane_count);
+}
+
 static struct intel_encoder *
 intel_ddi_get_crtc_encoder(struct drm_crtc *crtc)
 {
@@ -1269,9 +1280,7 @@ static void intel_ddi_pre_enable(struct intel_encoder *intel_encoder)
   struct intel_digital_port *intel_dig_port =
    enc_to_dig_port(encoder);
 
-  intel_dp->DP = intel_dig_port->saved_port_bits |
-          DDI_BUF_CTL_ENABLE | DDI_BUF_EMP_400MV_0DB_HSW;
-  intel_dp->DP |= DDI_PORT_WIDTH(intel_dp->lane_count);
+  intel_ddi_init_dp_buf_reg(intel_encoder);
 
   intel_dp_sink_dpms(intel_dp, DRM_MODE_DPMS_ON);
   intel_dp_start_link_train(intel_dp);
diff --git a/drivers/gpu/drm/i915/intel_drv.h b/drivers/gpu/drm/i915/intel_drv.h
index 70376b1..52986be 100644
--- a/drivers/gpu/drm/i915/intel_drv.h
+++ b/drivers/gpu/drm/i915/intel_drv.h
@@ -717,6 +717,7 @@ void intel_ddi_fdi_disable(struct drm_crtc *crtc);
 void intel_ddi_get_config(struct intel_encoder *encoder,
      struct intel_crtc_config *pipe_config);
 
+void intel_ddi_init_dp_buf_reg(struct intel_encoder *encoder);
 
 /* intel_display.c */
 const char *intel_output_name(int output);
-- 
1.7.1